lib/kernel/doc/guides/eep48_chapter.md

```diff
@@ -145,8 +148,11 @@ information to each entry. This EEP documents the following metadata keys:
 - **`since := binary()`** - a binary representing the version such entry was
   added, such as `<<"1.3.0">>` or `<<"20.0">>`.
 
-- **`edit_url := binary()`** - a binary representing a URL to change the
-  documentation itself.
+- **`source := binary()`** - the location of the source file for this module.
+  Applies only to the module metadata.
+
+- **`source_anno := erl_anno:anno()`** - annotation metadata of the source
```

Comment:
Range is usually more handy than position

Reply:
When is a range used and when is a position used? Remember this is external information, not information about when a file is currently open in the editor (especially because, when a file is opened, its locations may have all changed already anyway).

Reply:
Almost everything in LSP operates on ranges (e.g. [go to definition](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#textDocument_definition) is `(documentURI, position) -> (documentURI, range)`. If you search for Postion in the spec it's ~140 results vs 430 for Range + 80 for Location (uri and range)

Reply:
So in Elixir's case, it would go from `def ... end`? And in Erlang case, from the name until the dot? What happens when we only have a starting location, how are you filling it in? And, even if you had the range, would it actually be use by language servers? I never saw Go to Definition actually selecting the whole code. Which is also tricky if you have multiple clauses.

Reply:
It’s hard to answer. Either def to end or function name only. Maybe function name is more useful. Def to end use cases will most likely need parsing the source file anyway and we can get ranges from AST.
If the cursor is inside definition range editors do find references request so it definitely returns ranges e.g. in typescript VSCode.

Reply:
> If the cursor is inside definition range editors do find references request so it definitely returns ranges e.g. in typescript VSCode.

Right, but in this case, you do have the file open anyway, so you can parse the range. Do we need the range for any operation when the file itself won't be open?

Reply:
IMO we should stick with an `erl_anno` position as that is what is used in the `Anno` field of doc chunks.

Reply:
@garazdawi I talked to @lukaszsamson and here are the concerns:

1. One function can actually have multiple clauses, with macros, attributes, etc. between them. So perhaps we should store the entry for each clause. So should we name the field `source_annos` and have it as a list?

2. In the future, storing the full range may be necessary. However, I think it should be trivial to add `end_location` to `erl_anno` in case we need it in the future. It already supports metadata attributes such as `text` and `record`, and a new field should be ok.

So my proposals are:

1. Update this PR to use `source_path` and `source_annos`
2. Send a PR adding `end_location` and `set_end_location` to `erl_anno`

WDYT?

Reply:
For the `source_annos`, you can keep just one, if you don't want to keep them all, but we should support multiple if you want to.

Reply:
Oh, `erl_anno` already has `end_location` but computes it from text. We could also allow it to just manually store it too. Fantastic.

Reply:
I don't mind having `source_annos`, though I'm not sure I see where that would be useful. Would ExDoc render multiple links to the source? Would ElixirLS use it to give multiple possible goto definition entries?

Regarding `end_location` to `erl_anno`, I would not mind such an addition, but that impacts the compiler folks a lot more than me so we will have to bring them into the discussion :) But feel free to open a PR and we can continue that discussion there.

Reply:
The editor supports multiple entries, so I think we should consider that (given that's one of the main use cases) but in most cases we will have it as a single entry. I will submit a PR to add `erl_anno:end_location` and I will update this one based on the outcome of that. Be back soon. :)